src/reducers/errorReducer.js

```diff
@@ -0,0 +1,29 @@
+import produce from 'immer';
+
+const initialState = {};
+
+const errorReducer = produce((draft, action) => {
+  const { type, payload } = action;
+  const matches = /(.*)_(SUCCESS|ERROR)/.exec(type);
+
+  // not a *_SUCCESS / *_ERROR actions, so we ignore them
+  if (!matches) return;
+
+  const [, requestName, requestState] = matches;
+
+  switch (requestState) {
+    case 'ERROR':
+      draft[requestName] = payload?.message ?? '';
```

Comment:
This means we store nothing in case of an error without a message?

Or will we have to make sure that we always dispatch a `*_ERROR` action with a message? 

I'm trying to understand the flow of error data here... when an error occurs, it will hit the nearest error boundary right, and then we dispatch a general error from that boundary? And then the component that the error boundary is rendering looks into the state to show some error information?

Reply:
> This means we store nothing in case of an error without a message?

> Or will we have to make sure that we always dispatch a `*_ERROR` action with a message?
> 

We will always want store something. This is just a guard against us forgetting to add the error to the action.

> I'm trying to understand the flow of error data here... when an error occurs, it will hit the nearest error boundary right, and then we dispatch a general error from that boundary? And then the component that the error boundary is rendering looks into the state to show some error information?

The error boundary will dispatch nothing, but in some cases we might want to dispatch something. Error boundaries is more about showing a fallback UI, and logging to slack. 

We have to think about error boundaries as consecuences of dispatching an error, not as error dispatchers.

Reply:
Ok, it's just in some cases there will be errors that happen with no `*_ERROR` action. Just like uncaught errors and mistakes in components themselves. 

And I think that's the more common case that an error boundary will become visible. 

So I'm trying to understand in what situations and what components will try to render data from this error reducer. 

But it's ok, I don't see the big picture yet.